interface/mmal/vc/mmal_vc_msgs.h

```diff
@@ -515,7 +612,7 @@ static inline void mmal_vc_buffer_header_to_msg(mmal_worker_buffer_from_host *ms
    msg->buffer_header.pts           = header->pts;
    msg->buffer_header.dts           = header->dts;
    msg->buffer_header.alloc_size    = header->alloc_size;
-   msg->buffer_header.data          = header->data;
+   msg->buffer_header.data          = (uintptr_t)header->data;
    msg->buffer_header_type_specific = *header->type;
 }
 
```

Comment:
This whole commit is a bit TL;DR for me to have fully analysed it, but it all looks plausible.

Reply:
Sorry, I had started with bits of it as some individual commits, but hadn't been careful enough about squashing fixups into the right commit, so squahsed the whole lot into one.
It's mainly using a lookup table for client contexts and component contexts - I can try splitting the various bits out into independent commits if you really want it. There's then the changes to the RPC structures as well with some casts.

Annoyingly the contexts get allocated (and freed) from mmal_vc_api.c, but all the actual lookups are in mmal_vc_client, hence exporting only some of the functions.

Reply:
Pasting that into the commit message is probably enough.

Reply:
OK, I'll update the commit message with more detail of what exactly has been changed and why.